### ISSUE TITLE
MeshCentral inside iFrame with AzureAD authorization

### ISSUE THREAD

**Issue by AntonAndreevichMoroz** [Ylianst/MeshCentral#2581]
I don’t know how to qualify this request. As a Question or as a BUG.
I am trying to embed MeshCentral into our system via an iframe. Naturally, you want to use SSO from AzureAD so that you don't have to log in to each system. But there is such a problem that when using a local account or Login Tokens, everything works. But if you use the authorization button (link https://rm.realsd.ru/auth-azure), then there is a redirect back to the authorization page, i.e. it is not possible to log in and we constantly see the login window.
In the server trace, I see that, in fact, authorization takes place and MeshCentral sees my login.
handleStrategyLogin: {"sid": "~ azure: name@domain.ru", "name": "User Name", "strategy": "azure"}
Is this a bug or is there something wrong with my settings? Can I do something about it?

**Comment by Ylianst** [comment]
So, should I understand that everything works without having MeshCentral within an iframe? In general, SSO do not allow authentication of pages within iframes because of security. The parent HTML on top of the iframe has too much power over the iframe page that SSO services would likely not allow this. If I am not correct on your situation, let me know.

**Comment by AntonAndreevichMoroz** [comment]
I found some information on the internet. Some sites indicate that if you add the parameter prompt = none to the request, then this allows you to use embedding. This, according to reviews, works if the authorization has already been passed and you just need to get or update the token.
Here is some official information
https://docs.microsoft.com/en-us/azure/active-directory/develop/v2-oauth2-implicit-grant-flow

**Comment by Ylianst** [comment]
Ok. Not sure if you are asking for anything MeshCentral related. This is probably something specific to your SSO provider.

**Comment by AntonAndreevichMoroz** [comment]
Perhaps I jumped to my conclusions. I found in the code where the link to the request for Azure is generated, made minor changes, the parameter appeared in the request, but it did not help. Probably too much concerned about security while embedding other sites and greatly curtailed all the possibilities. I will close this request. If there is new information, I will apply again.
Thank you
